### PR TITLE
ENH: Reorganize and add features

### LINKED ISSUE #60: Ignoring directories
Hello,

I would like to ignore everything that's in (or anywhere below) a specific directory. This currently doesn't seem to be supported, though.
Would you mind implementing it?

Thanks for the useful tool!

### LINKED ISSUE #63: please set returncode according to file(s) OK or not
no errors = 0 and errors found should give >0.

for now it always succeeds:
```
user@box:~# ./codespell.py -i 0 $FILE
(several mistakes)
user@box:~# echo $?
0
```
but i exspect e.g. '1'

### LINKED ISSUE #64: Python2 support
Would the devs be interested in me adding support for Python2? I've worked on a number of packages with support for Python2 and Python3, so I don't expect it would be too bad.

### PR DESCRIPTION
**Pull request by Eric89GXL**
A bunch of changes here. I recognize this is an invasive set of changes, so I can split into multiple PRs if it helps, but I wanted to show them all for discussion:

1. Make it Py2 compatible by using `codecs.open` (#64)
2. Adds returncode for number of errors/warnings. (#63)
3. Does matching with directories in ignore. (#60)
4. Reorganizes to use standard `distutils` for installation (enables #38).
5. Increments to next version number.

Closes #64.
Closes #63.
Closes #60.

I'm a bit concerned that there could be un-caught regressions in here because there aren't any tests. Should I write some? I might be able to get Travis-CI integration going if you enable it @lucasdemarchi.

### CODE REVIEW

File: codespell_lib/codespell.py [hunk 1]
```diff
@@ -314,10 +312,9 @@ def is_hidden(filename):
 def is_text_file(filename):
     with open(filename, mode='rb') as f:
         s = f.read(1024)
-        if 0 in s:
-            return False
-
-        return True
+    if b'\x00' in s:
```
Comment (Eric89GXL) [inline codespell_lib/codespell.py]: I think this is equivalent, since on Py3k I get:
```
>>> 0 in b'\x00'
True
```

**Comment by lucasdemarchi** [comment]
> Should I write some? I might be able to get Travis-CI integration going if you enable it @lucasdemarchi.

One thing I was doing in the past was given the same source code do parse before and after the changes. Automating this to run on travis would be nice

**Comment by lucasdemarchi** [comment]
> A bunch of changes here. I recognize this is an invasive set of changes, so I can split into multiple PRs if it helps, but I wanted to show them all for discussion:

not sure if it needs to be in separate PRs. Separate commits would be sufficient.

**Comment by lucasdemarchi** [comment]
so... you already detail the changes in the PR message. These would be reasonable commits

**Comment by Eric89GXL** [comment]
As a first step then I'd like to do some tests and travis, can you enable
it for this repo on the travis site?

**Comment by lucasdemarchi** [comment]
https://travis-ci.org/lucasdemarchi/codespell

done

**Comment by lucasdemarchi** [comment]
I also added you as a collaborator to make things easier for you.

**Pull request closed without merging**